Nimporter fails to import due to undefined symbol "__builtin_ssubll_overflow" when using GCC on CentOS 7.5

So, I got this really ugly stack trace when trying to import the basic `nim_math.nim` example code into my python project. The relevant sections of the project tree look like:

```
root/
    nim/
        nim_math.nim
        __pycache__/
            nim_math.nim.hash
            nim_math.so
    src/
        app.py
```

And the relevant contents of `app.py` are: 

```
...
import nimporter
import nim.nim_math
...
```

Here's some more data about the environment that isn't available in the stack trace: I'm using a CentOS 7.5 base docker image with GCC version 4.8.5. I've also replicated this same failure with the same docker image on GCC versions 7.x and 8.x.

```
/pyenv/versions/3.8.0/envs/my-new-project/lib/python3.8/site-packages/nimporter.py:1117: in find_spec
    return Nimporter.import_nim_code(fullname, path, library=False)
/pyenv/versions/3.8.0/envs/my-new-project/lib/python3.8/site-packages/nimporter.py:819: in import_nim_code
    cls.__validate_spec(spec)
/pyenv/versions/3.8.0/envs/my-new-project/lib/python3.8/site-packages/nimporter.py:887: in __validate_spec
    raise NimporterException(error_message) from import_error
E   nimporter.NimporterException: Error importing /app/nim/__pycache__/nim_math.so
E   Error Message:
E
E       /app/nim/__pycache__/nim_math.so: undefined symbol: __builtin_ssubll_overflow
E
E   Python Version:
E
E       3.8.0 (default, Dec 16 2019, 22:06:06) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
E
E   Nim Version:
E
E       Nim Compiler Version 1.2.0 [Linux: amd64]
E
E   CC Version:
E
E       <Error getting version>
E
E   Installed CCs:
E
E       {'gcc': PosixPath('/usr/bin/gcc')}
E
E   Please help improve Nimporter by opening a bug report at: https://github.com/Pebaz/nimporter/issues/new and submit the above information along with your description of the issue.
```

I'm not sure if the issue is with nimporter, nimpy, the nim compiler, or gcc itself. It _looks_ like this is an issue with the compiled output, so nimporter might be off the hook here. If that's the case, I'll have to file this same issue one layer deeper down the stack - which I guess would involve pestering the folks who develop nimpy.

I'm hoping this is an easy fix though - maybe there's some compiler switch I'm missing that could make this all go away. Thanks for the help (and the awesome project!)

--- a/README.md
+++ b/README.md
@@ -245,6 +245,19 @@ automatically, use the `github_actions_template.yml` template found in the
 Actions runner to build, package, and deploy your library on Windows, MacOS, and
 Linux automatically when you create a new "Release" is created.
 
+### Usage with Docker
+
+Nimporter can easily be used within a Docker container. To prevent the need for
+a Nim compiler toolchain to be installed into the container to run Nim code, you
+can pre-compile all of your extensions and copy the resulting artifacts into the
+container. This process is roughly as follows:
+
+1. Create a project that uses Python and Nim
+2. Run `nimporter compile` to recursively-compile all extensions in the project
+3. Ensure that in your Dockerfile that the `__pycache__` directories are
+   included as they will contain the Nim shared objects as well as the Nimporter
+   hash files to prevent a recompilation.
+
 ## Nimporter Command Line Interface
 
 Nimporter provides a CLI that you can use to easily clean all cached build and
@@ -275,6 +288,16 @@ $ nimporter build mylib --dest .
 
 # Although you can specify a Nim library's source file, please don't
 $ nimporter build mylib/mylib.nim
+```
+
+The Nimporter CLI can also precompile all extensions within a project without
+needing to run the project. This is useful in situations where you do not want
+to package your application using a `setup.py` (such as a zip file) or for use
+within Docker containers.
+
+```bash
+# Recursively compile all Nim extension modules and libraries:
+$ nimporter compile
 ```
 
 Finally, the CLI has provisions for quickly bundling your project into a source

--- a/nimporter_cli.py
+++ b/nimporter_cli.py
@@ -2,7 +2,7 @@
 Iterates through all sub directories and removes any build artifacts and hashes.
 """
 
-import sys, os, pathlib, argparse, tempfile, shutil, subprocess
+import sys, os, pathlib, argparse, tempfile, shutil, subprocess, time
 from nimporter import NimCompiler, Nimporter
 
 def clean(dir=pathlib.Path()):
@@ -40,8 +40,22 @@ def main(cli_args=None):
     parser = argparse.ArgumentParser(description='Nimporter CLI')
     subs = parser.add_subparsers(dest='cmd', required=True)
 
-    subs.add_parser('clean')
-    build = subs.add_parser('build')
+    # Clean command
+    subs.add_parser(
+        'clean',
+        help=(
+            'Run in project root to recursively remove all Nimporter-specific '
+            'build artifacts and hash files'
+        )
+    )
+
+    # Build command
+    build = subs.add_parser(
+        'build',
+        help=(
+            'Builds a Nim module/library into an importable Python extension'
+        )
+    )
     build.add_argument(
         'source',
         type=pathlib.Path,
@@ -53,10 +67,26 @@ def main(cli_args=None):
         help='the folder to store the build artifact'
     )
 
-    bundle_parser = subs.add_parser('bundle')
+    # Bundle command
+    bundle_parser = subs.add_parser(
+        'bundle',
+        help=(
+            'Convenience command for running: python setup.py sdist/bdist_wheel'
+        )
+    )
     bundle = bundle_parser.add_subparsers(dest='exp', required=True)
     bin_ = bundle.add_parser('bin')
     src = bundle.add_parser('src')
+
+    # Compile command
+    compile_ = subs.add_parser(
+        'compile',
+        help=(
+            'Clean project and then recurse through and build all Nim '
+            'modules/libraries'
+        )
+    )
+
     args = parser.parse_args(cli_args or sys.argv[1:])
 
     if args.cmd == 'clean':
@@ -142,6 +172,38 @@ def main(cli_args=None):
 
             elif args.exp == 'src':
                 subprocess.Popen(f'{pyexe} setup.py sdist'.split()).wait()
+
+    elif args.cmd == 'compile':
+        clean()
+
+        CTM = lambda: round(time.time() * 1000)
+        start = CTM()
+        extensions = Nimporter._find_extensions(pathlib.Path())
+
+        for extension in extensions:
+            is_lib = extension.is_dir()
+
+            print(
+                f'Building Extension {"Lib" if is_lib else "Mod"}: '
+                f'{extension.name}'
+            )
+
+            NimCompiler.compile_nim_code(
+                extension.absolute(),
+                NimCompiler.build_artifact(extension.absolute()),
+                library=is_lib
+            )
+
+            if is_lib:
+                Nimporter.update_hash(extension / (extension.name + '.nim'))
+            else:
+                Nimporter.update_hash(extension)
+
+        print('Done.')
+        print(
+            f'Built {len(extensions)} Extensions In '
+            f'{(CTM() - start) / 1000.0} secs'
+        )
 
     return 0
 

